### ISSUE TITLE
Semantic Highlighting

### ISSUE THREAD

**Issue by ChrisCoffey** [ensime/ensime-vim#199]
The emacs client & server support semantic highlighting. Seeing how this is a pretty useful feature, it would be great if we supported it as well.

**Comment by jvican** [comment]
Are you up for taking this @ChrisCoffey?

**Comment by ChrisCoffey** [comment]
Not in the next couple weeks.

**Comment by jvican** [comment]
No prob, there's no hurry. And also, feel free to unassign yourself from any issue if you see you won't be able to do it. Cheers!

**Comment by ktonga** [comment]
I don't think this is possible, Vim doesn't provide the needed API.
See neovim/neovim#719 for a long and nutritive discussion about the subject.

**Comment by ChrisCoffey** [comment]
I disagree. This is quite a large task, but I think this is perfectly doable using `matchaddpos()` to link up the buffer text with the ast. The 8 pos cap on `matchaddpos()` will present an issue for frequently used values, but again I think that can be worked around.
